isis/scripts/isisVarInit.py

```diff
@@ -20,12 +22,50 @@
 #
 # SPDX-License-Identifier: CC0-1.0
 
+# This is the full path to this environment:
+CONDA_PREFIX = os.environ["CONDA_PREFIX"]
+ENVS_DIR = Path(CONDA_PREFIX).parent
+THIS_ENV = Path(CONDA_PREFIX).name
+
+
+def get_conda_envs() -> List[Path]:
+    """Returns a list of paths to the found conda envs.
+
+    This method is used because
+    (1) there is no API for `conda env`,
+    (2) `subprocess.check_output` and `sh.conda` are not producing the same output.
+    """
+    print(f"Searching for other conda envs in {ENVS_DIR} ...")
+    paths = [d for d in ENVS_DIR.iterdir() if d.is_dir()]
+    return [p.name for p in paths if p.name != THIS_ENV]
+
+
+def get_envs_to_init() -> List[Path]:
+    """Find conda envs and ask user which ones should initialize ISIS."""
+    env_names = get_conda_envs()
+    print("Found the following other conda envs:")
+    for i, env in enumerate(env_names):
+        print(i, env)
+    print("Which envs should receive the ISIS path initialization?")
+    print("(Press RETURN for skipping.)")
+    answer = ""
+    answer = input("Comma-separated list of env indexes (e.g. 1,2,4):")
+    if not answer:
+        return []
+    try:
+        envs_to_init = [env_names[int(i)] for i in answer.split(",")]
+    except ValueError:
+        raise ValueError(
+            "Could not parse your answer.\nDid you use a comma-separated list?"
+        )
+    return envs_to_init
+
 
 def mkdir(p: Path) -> str:
-    """Returns a string with a message about the creation or existance
-    of the path, *p*."""
+    """Create directory `p` if not exists, and notify user when created."""
+
     if p.exists():
-        return f"{p} exists, don't need to create."
+        return p
```

Comment:
This function now sometimes returns a string, and sometimes returns a Path, which is weird (and violates the return signature at the very least).  It should always return a string (like it did) or always return a Path.  Pick one.